frontend/src/scenes/insights/insightSceneLogic.tsx

```diff
@@ -494,9 +495,11 @@ export const insightSceneLogic = kea<insightSceneLogicType>([
             if (q) {
                 const validQuery = typeof q === 'string' ? parseDraftQueryFromURL(q) : q
                 if (validQuery) {
-                    validatingQuery = true
                     if (initial) {
+                        validatingQuery = true
                         actions.upgradeQuery(validQuery)
+                    } else if (currentInsightId !== insightId) {
+                        queryFromUrl = typeof q === 'string' ? parseDraftQueryFromURL(q) : null
```

Comment:
The problem here is that `insightId` might equal `"new"`, and you might be comparing "new" with "new" even though technically this should fire.

An alternative that might work is to use `method`. Whenever we set the `#q` ourselves, we set it with `router.actions.replace`. My biggest worry is that we'll get stuck in a loop, where the URL calls the action, and the action calls the URL. One thing that can avoid this is that we only read the URL, if the user navigated to it with a back/forward `pop` action, or if a link was clicked (`push`)... and explicitly avoiding all `replace` actions.

So while I haven't tested it, checking for `method !== 'replace'` here might work well... 🤔